File: backend/src/main/java/com/warlocksai/videoclub/model/Director.java
package com.warlocksai.videoclub.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Director {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;

    @OneToMany(mappedBy = "director", cascade = CascadeType.ALL)
    private List<Pelicula> peliculas = new ArrayList<>();

    public Director() {
    }

    public Director(String nombre) {
        this.nombre = nombre;
    }

    public Long getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public List<Pelicula> getPeliculas() {
        return peliculas;
    }
}

File: frontend/src/app/models/director.ts
import { Pelicula } from './pelicula';

export interface Director {
  id: number;
  nombre: string;
  totalPeliculas: number;
}

export interface DirectorDetalle {
  id: number;
  nombre: string;
  peliculas: Pelicula[];
}

export interface CrearDirector {
  nombre: string;
}
